superset/assets/javascripts/explore/exploreUtils.js

```diff
@@ -1,13 +1,30 @@
 /* eslint camelcase: 0 */
 import URI from 'urijs';
 
+export function getChartKey(explore) {
+  const slice = explore.slice;
+  return slice ? ('slice_' + slice.slice_id) : 'slice';
+}
+
+export function getAnnotationJsonUrl(slice_id, form_data, isNative) {
```

Comment:
NOTE: there's a PR by @graceguo-supercat  moving things to use `POST`, depending on what merges first there might be some intricate merge conflicts here

Reply:
I would love to get this in before that refactor. I just went through 2 days of resolving merge conflicts because the JavaScript got refactored. And I had this out for 4 month.

Reply:
Agreed, let's merge this first.